### PR TITLE
Feature/bcpp 545 create announcement

### PR DESCRIPTION
**Pull request by mykytabelimenko**

### CODE REVIEW

File: plugins/announcements/src/helpers/announcement-helper.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import bugImage from '../assets/bug.png';
+import generalImage from '../assets/general.png';
+import immediateImage from '../assets/immediate.png';
+import releaseFeatureImage from '../assets/release-feature.png';
+import announcementImage from '../assets/announcement.png';
+
+export const getActionLinkTitle = (categorySlug: string) => {
```
Comment (novellat) [inline plugins/announcements/src/helpers/announcement-helper.ts:7]: just thinking, how about making category slug an enum? or at least string enum? 

export type categorySlug = 'release-feature' | 'immediate' | ....

File: plugins/announcements-backend/src/service/router.ts [hunk 2]
```diff
@@ -144,7 +144,7 @@ export async function createRouter(
           ...initialAnnouncement,
           ...{
             title: req.body.title,
-            excerpt: req.body.excerpt,
+            link: req.body.link,
```
Comment (novellat) [inline plugins/announcements-backend/src/service/router.ts:147]: I see you're renaming exceprt to a link, but I see no explanation in the jira ticket, or PR summary, or even in PR comment. Can you explain, why are you doing it? It's useful for me as a reviewer to understand the context

File: plugins/announcements/src/helpers/announcement-helper.ts (hunk 1 above)
Comment (novellat) [inline plugins/announcements/src/helpers/announcement-helper.ts:7]: why is ti taking a string, not a category slug?

File: plugins/announcements/src/helpers/announcement-helper.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    case 'immediate':
+      return { image: immediateImage, icon: '🚨' };
+    case 'bug-alert':
+      return { image: bugImage, icon: '🐞' };
+    default:
+      return { image: generalImage, icon: '🔔' };
+  }
+};
+
+export const validateCategorySlug = (slug: unknown): slug is CategorySlug => {
```
Comment (novellat) [inline plugins/announcements/src/helpers/announcement-helper.ts:46]: maybe i'd rename to isCategorySlug, but it's a matter of taste

File: plugins/announcements/src/helpers/announcement-helper.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+interface CategoryInfo {
+  image: string;
+  icon: string;
+}
+
+export const getAnnouncementInfo = (slug: CategorySlug): CategoryInfo => {
+  switch (slug) {
+    case 'announcement':
+      return { image: announcementImage, icon: '📣' };
```
Comment (novellat) [inline plugins/announcements/src/helpers/announcement-helper.ts:32]: i still find it hilaroous how unicode images can be a part of code hehe

File: plugins/announcements/src/helpers/announcement-helper.ts (hunk 3 above)
Comment (novellat) [reply plugins/announcements/src/helpers/announcement-helper.ts:46]: where are yo uusing this?

File: plugins/announcements/src/components/EditAnnouncementPage/EditAnnouncementPage.tsx [hunk 5]
```diff
@@ -38,19 +53,32 @@ export const EditAnnouncementPage = (props: EditAnnouncementPageProps) => {
...
+  };
+
   if (loading) {
     content = <Progress />;
   } else if (error) {
     content = <Alert severity="error">{error.message}</Alert>;
   } else {
-    title = `Edit "${value!.title}" – ${title}`;
-    content = <AnnouncementForm initialData={value!} onSubmit={onSubmit} />;
+    title = `Edit ${updatedValue?.category.title} / ${value!.title} – ${title}`;
```
Comment (novellat) [inline plugins/announcements/src/components/EditAnnouncementPage/EditAnnouncementPage.tsx:65]: interesting, I was expecting value to be unwrapped in this branch

File: plugins/announcements/src/components/ReadMore/ReadMore.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  detailsPageUrl: string;
+}
+
+export const ReadMore: React.FC<ReadMoreProps> = ({
+  text,
+  maxLength,
+  detailsPageUrl,
+}) => {
+  const truncatedText =
+    text.length > maxLength ? `${text.substring(0, maxLength)}...` : text;
```
Comment (novellat) [inline plugins/announcements/src/components/ReadMore/ReadMore.tsx:16]: just an idea, MAYBE you can just use css.  text-overflow: ellipsis;

File: .env [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export POSTGRES_HOST=127.0.0.1
```
[the remark this answers is not in the document]
Comment (mykytabelimenko) [reply .env:1]: Good point, thx

File: plugins/announcements-backend/src/service/router.ts (hunk 2 above)
Comment (mykytabelimenko) [reply plugins/announcements-backend/src/service/router.ts:147]: Yes, pardon me :)    
The main reason was, that we don't need excerpt, but we need link field, that's why I think renaming would be more easier to do.

**Pull request merged**